src/praisonai-agents/praisonaiagents/memory/adapters/perseus_vault_adapter.py

```
@@ -0,0 +1,342 @@
+        A plain ``readline()`` blocks forever if the child accepts the request
+        but never emits a newline, which would hang every subsequent memory
+        call while the lock is held. Reading on a daemon thread lets the
+        deadline actually fire and surface a ``TimeoutError``.
+        """
+        assert self._proc and self._proc.stdout
+        result: List[Optional[str]] = [None]
+
+        def _read() -> None:
+            try:
+                result[0] = self._proc.stdout.readline()
+            except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)